lightning/src/chain/channelmonitor.rs

```diff
@@ -1984,6 +2046,31 @@ impl<Signer: Sign> ChannelMonitorImpl<Signer> {
 		where B::Target: BroadcasterInterface,
 		      F::Target: FeeEstimator,
 					L::Target: Logger,
+	{
+		self.update_best_block(header, height, &logger);
+		self.transactions_confirmed(header, txdata, height, broadcaster, fee_estimator, &logger)
+	}
+
+	fn update_best_block<L: Deref>(&mut self, header: &BlockHeader, height: u32, logger: &L)
+	where L::Target: Logger {
+		let block_hash = header.block_hash();
+		log_trace!(logger, "New best block {} at height {}", block_hash, height);
```

Comment:
All of the height-based triggers should fire if only `update_best_block` is called but `transactions_confirmed` isn't at a given height.

Reply:
How would you define "height-based triggers"? AFAICT, there are places where only height is considered:

https://github.com/rust-bitcoin/rust-lightning/blob/cb442f8271f21d25fb50b82c215bf2dd9cf5bfa5/lightning/src/chain/channelmonitor.rs#L2144-L2163

And other places where height is at first only considered:

https://github.com/rust-bitcoin/rust-lightning/blob/cb442f8271f21d25fb50b82c215bf2dd9cf5bfa5/lightning/src/chain/channelmonitor.rs#L2129-L2143

But which the produced `claimable_outpoints` are used later along with both `txn_matched` and `height`

https://github.com/rust-bitcoin/rust-lightning/blob/cb442f8271f21d25fb50b82c215bf2dd9cf5bfa5/lightning/src/chain/channelmonitor.rs#L2165

But then digging into `update_claims_view`, sometimes only `height` is needed while other times `tx_matched` is also needed.

Just want some clarification before pursuing this as it seems there may be a lot to untangle.

Reply:
Yea, I do think there's gonna be some stuff to untagle, sadly. I think both of those cases, indeed, need to be untangled, which will definitely require a similar refactor/untangling in `OnchainTxHandler`, though it may not be too crazy in there.

Reply:
This all fell in place in a much cleaner way than I had imagined. AFAICT, `OnchainTxHandler::update_claims_view` works with or without any transactions, which makes things easy. So the `ChannelMonitor` refactoring was very straight-forward.

Could you give this a quick pass in case I'm missing something crucial? If all looks good, I can add wrappers in `ChainMonitor`.

Reply:
At first glance this looks good. Obviously would be nice to test, though.